### PR TITLE
[MODULAR: FULLY, EVEN] Fixes & Adjusts Snail Shell Capacity

### PR DESCRIPTION
**Pull request by Nerev4r**
## About The Pull Request
Hi! As it turns out, I am an idiot and accidentally made snail shells, on roundstart, able to hold 'huge' sized items; apparently, four shards of gibtonite.
This is terrible.
So I have set them back to duffelbag level PROPERLY as intended, however, I have added just a little bit of compensation. Now, snails will be able to slap their shells with a bluespace core in order to transform them into a vaulted 'shell of holding.'

image
image
image
Great pains were taken to both create the fact that this overlay can exist, and make it look good.

## How This Contributes To The Skyrat Roleplay Experience
Storage woes being resolved is nice. Also, you're trading the bag of holding for literally not having anything else on your back slot; remember they can't use MODsuits.

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
 
image
  
</details>

## Changelog
:cl:
fix: Snails no longer have the ability to carry huge-sized items such as Kinetic Crushers on roundstart in their shells.
add: Snails can now slap their shells with a bluespace core to turn them into shells of holding.
/:cl:

### CODE REVIEW

File: modular_skyrat/master_files/code/modules/mob/living/carbon/human/species_type/snail.dm [hunk 1]
```diff
@@ -1,6 +1,42 @@
+#define SHELL_TRANSPARENCY_ALPHA 90
+
+/obj/item/storage/backpack/snail
+	var/storage_core = FALSE
```
Comment (softcerv) [inline modular_skyrat/master_files/code/modules/mob/living/carbon/human/species_type/snail.dm:5]: make an autodoc comment explaining what this variable does please!

File: modular_skyrat/master_files/code/modules/mob/living/carbon/human/species_type/snail.dm [hunk 2]
```diff
@@ -1,6 +1,42 @@
...
+		storage_core = TRUE
+		qdel(core)
+		emptyStorage()
+		create_storage(max_specific_storage = WEIGHT_CLASS_GIGANTIC, max_total_storage = 35, max_slots = 30, storage_type = /datum/storage/bag_of_holding)
+		atom_storage.allow_big_nesting = TRUE
+		name = "snail shell of holding"
+		user.update_worn_back()
+		update_appearance()
+		return
+	return ..()
```
Comment (softcerv) [inline modular_skyrat/master_files/code/modules/mob/living/carbon/human/species_type/snail.dm]: This is a case where you'd want to have it be an early return, send me a message on discord if you don't know what I mean by that.

File: modular_skyrat/master_files/code/modules/mob/living/carbon/human/species_type/snail.dm [hunk 3]
```diff
@@ -1,6 +1,42 @@
...
+
 /obj/item/storage/backpack/snail/Initialize(mapload)
 	. = ..()
-	atom_storage.max_specific_storage = 30
+	atom_storage.max_total_storage = 30
+
+/obj/item/storage/backpack/snail/attackby(obj/item/core, mob/user)
+	if(istype(core, /obj/item/assembly/signaler/anomaly/bluespace))
+		to_chat(user, span_notice("You insert [core] into your shell, and it starts to glow blue with expanded storage potential!"))
+		playsound(src.loc, 'sound/machines/click.ogg', 50, TRUE)
```
Comment (softcerv) [inline modular_skyrat/master_files/code/modules/mob/living/carbon/human/species_type/snail.dm]:
```suggestion
		playsound(src, 'sound/machines/click.ogg', 50, TRUE)
```

File: modular_skyrat/master_files/code/modules/mob/living/carbon/human/species_type/snail.dm (hunk 2 above)
Comment (vinylspiders) [inline modular_skyrat/master_files/code/modules/mob/living/carbon/human/species_type/snail.dm]:
```suggestion
/obj/item/storage/backpack/snail/attackby(obj/item/core, mob/user)
	if(!istype(core, /obj/item/assembly/signaler/anomaly/bluespace))
		return ..()

	to_chat(user, span_notice("You insert [core] into your shell, and it starts to glow blue with expanded storage potential!"))
	playsound(src, 'sound/machines/click.ogg', 50, TRUE)
	add_filter("bluespace_shell", 2, list("type" = "outline", "color" = COLOR_BLUE_LIGHT, "size" = 1))
	storage_core = TRUE
	qdel(core)
	emptyStorage()
	create_storage(max_specific_storage = WEIGHT_CLASS_GIGANTIC, max_total_storage = 35, max_slots = 30, storage_type = /datum/storage/bag_of_holding)
	atom_storage.allow_big_nesting = TRUE
	name = "snail shell of holding"
	user.update_worn_back()
	update_appearance()
```

Comment (vinylspiders) [reply modular_skyrat/master_files/code/modules/mob/living/carbon/human/species_type/snail.dm]: I think this is what @softcerv was getting at if you want to be lazy. : )

File: modular_skyrat/master_files/code/modules/mob/living/carbon/human/species_type/snail.dm (hunk 1 above)
Comment (vinylspiders) [inline modular_skyrat/master_files/code/modules/mob/living/carbon/human/species_type/snail.dm:5]:
```suggestion
	/// Whether or not a bluespace anomaly core has been inserted
	var/storage_core = FALSE
```

**Comment by tf-4** [comment]
I feel like not being able to use bags of holding is part of the downside of the race

**Comment by Nerev4r** [comment]
> I feel like not being able to use bags of holding is part of the downside of the race

I don't think it'd be a huge negative for them to be able to, given that bluespace cores are not exactly a common resource to have laying around (they aren't super used when they are either,) and they still can't use MODsuits or any other back slot item like sprayers. Not being able to use the most common spacesuit (which I intend to never remedy outside of that one antag belt slotted MOD) is far more damaging; and there's some Compelling™ things you could do with this. Get the HoS to buy you a bluespace core and be a designated ammo carrier, or be a mobile font of resources for Engineering. It's kewl.

**Comment by tf-4** [comment]
good enough reasoning for me

**Review by tf-4 — APPROVED** [review]

**Pull request merged**